lib/iris/tests/unit/fileformats/netcdf/test_Saver.py

```diff
@@ -1068,20 +1068,12 @@ def setUp(self):
         self.exp_emsg = "Cell measures with missing data are not supported."
 
     def test_masked_data__insitu(self):
-        # Test that the error is raised in the right place.
+        # Test that the error is not being raised.
         with self.temp_filename(".nc") as nc_path:
             saver = Saver(nc_path, "NETCDF4")
-            with self.assertRaisesRegex(ValueError, self.exp_emsg):
-                saver._create_generic_cf_array_var(
-                    self.cube, self.names_map, self.cm
-                )
-
-    def test_masked_data__save_pipeline(self):
-        # Test that the right error is raised by the saver pipeline.
-        with self.temp_filename(".nc") as nc_path:
-            with Saver(nc_path, "NETCDF4") as saver:
-                with self.assertRaisesRegex(ValueError, self.exp_emsg):
-                    saver.write(self.cube)
+            saver._create_generic_cf_array_var(
```

Comment:
There is a problem with this, because the previous idea was to do the minimum setup to enable us to call this routine + have the check fail.
Unfortunately, to get the routine to **_succeed_** we must do rather more setup on the file first.
So I added these lines, which do at least make the call succeed ...
```
    # Do minimum of saver.write to let us to create the var successfully
    _, cube_dimensions = saver._get_dim_names(self.cube)
    saver._create_cf_dimensions(self.cube, cube_dimensions)
```

But then I think it's not totally clear what this should be testing **_for_**.  
My instinct was to add more, so I initially added some rather overkill asserts to check that the array is encoded as expected, like this ...
```
    # Check that data is written with masked points + default fill-value
    file_var = saver._dataset.variables['cell_area']
    assert '_FillValue' not in file_var.ncattrs()
    fill_value = nc.default_fillvals['i8']
    file_array = file_var[:]
    assert np.all(file_array.data == [fill_value, 1, fill_value])
    assert np.all(file_array.mask == [True, False, True])
```
But on closer inspection, this is actually what **any** masked save should be doing, and there is now nothing in it which is specific to cell-measures !!

So, it might seem more logical to test the operation of `saver._add_cell_measures`.  
And maybe all we really need to do is check that `_add_inner_related_vars` or `_create_generic_cf_array_var` is called by `_add_cell_measures`.
But unfortunately that requires even _more_ prior setup of the file before it can function ... and it still isn't actually testing anything specific to cell-measures.

Fundamentally, I think the 'setup problem'  is because the code itself is not structured to _enable_ testing isolated routines in this way.

Since cell-measures are basically now no different to anything else which gets passed into `_create_generic_cf_array_var`,
after much thought, I think I would just 
**remove this whole test !** -- does that sound crazy, or what ?